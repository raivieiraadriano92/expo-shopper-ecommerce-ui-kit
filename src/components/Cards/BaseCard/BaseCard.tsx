import { ComponentProps, FunctionComponent } from 'react'

import { Center } from 'native-base'

export const BaseCard: FunctionComponent<ComponentProps<typeof Center>> = ({
  children,
  ...props
}) => (
  <Center
    bg="white"
    borderRadius="2xl"
    style={{
      shadowColor: '#1b1956',
      shadowOffset: {
        width: 4,
        height: 0
      },
      shadowOpacity: 0.1,
      shadowRadius: 30,
      elevation: 24
    }}
    {...props}>
    {children}
  </Center>
)
