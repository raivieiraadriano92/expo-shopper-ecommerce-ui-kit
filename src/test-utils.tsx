import { FunctionComponent } from 'react'

import { render } from '@testing-library/react-native'

import { AppWrapper } from '../src/AppWrapper'

const Wrapper: FunctionComponent = ({ children }) => <AppWrapper>{children}</AppWrapper>

const customRender: typeof render = (ui, options) => render(ui, { wrapper: Wrapper, ...options })

// re-export everything
export * from '@testing-library/react-native'

// override render method
export { customRender as render }
