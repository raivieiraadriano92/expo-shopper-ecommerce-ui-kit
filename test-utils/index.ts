import { render } from '@testing-library/react-native'
import { AppWrapper } from 'src/AppWrapper'

const customRender: typeof render = (ui, options) => render(ui, { wrapper: AppWrapper, ...options })

// re-export everything
export * from '@testing-library/react-native'

// override render method
export { customRender as render }
