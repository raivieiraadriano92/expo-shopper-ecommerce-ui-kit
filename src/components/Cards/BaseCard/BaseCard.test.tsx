import { Text } from 'native-base'
import { render, waitFor } from 'test-utils'

import { BaseCard } from './BaseCard'

describe('<BaseCard />', () => {
  it('renders correctly', async () => {
    //@ts-ignore
    const { findByText, toJSON } = render(
      <BaseCard>
        <Text>testing</Text>
      </BaseCard>
    )

    await waitFor(() => findByText('testing'))

    expect(toJSON()).toMatchSnapshot()
  })
})
