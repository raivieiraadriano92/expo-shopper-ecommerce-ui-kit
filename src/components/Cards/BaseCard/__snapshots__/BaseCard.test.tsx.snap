// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseCard /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 16,
          "display": "flex",
          "justifyContent": "center",
        },
        Object {
          "elevation": 24,
          "shadowColor": "#1b1956",
          "shadowOffset": Object {
            "height": 0,
            "width": 4,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 30,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": undefined,
          "color": "#1f2937",
          "fontFamily": "Poppins_400Regular",
          "fontSize": 14,
          "fontStyle": undefined,
          "fontWeight": undefined,
          "letterSpacing": 0,
          "lineHeight": 21,
          "textDecorationLine": undefined,
        }
      }
    >
      testing
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
